include/libcyphal/presentation/presentation.hpp

```diff
@@ -0,0 +1,214 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef LIBCYPHAL_PRESENTATION_PRESENTATION_HPP_INCLUDED
+#define LIBCYPHAL_PRESENTATION_PRESENTATION_HPP_INCLUDED
+
+#include "presentation_delegate.hpp"
+#include "publisher.hpp"
+#include "publisher_impl.hpp"
+#include "subscriber.hpp"
+#include "subscriber_impl.hpp"
+
+#include "libcyphal/common/cavl/cavl.hpp"
+#include "libcyphal/transport/errors.hpp"
+#include "libcyphal/transport/msg_sessions.hpp"
+#include "libcyphal/transport/transport.hpp"
+#include "libcyphal/transport/types.hpp"
+#include "libcyphal/types.hpp"
+
+#include <cetl/cetl.hpp>
+#include <cetl/pf17/cetlpf.hpp>
+
+#include <utility>
+
+namespace libcyphal
+{
+namespace presentation
+{
+
+// TODO: docs
+class Presentation final : private detail::IPresentationDelegate
+{
+public:
+    Presentation(cetl::pmr::memory_resource& memory, IExecutor& executor, transport::ITransport& transport) noexcept
+        : memory_{memory}
+        , executor_{executor}
+        , transport_{transport}
+    {
+    }
+
+    /// @brief Makes a message publisher.
+    ///
+    /// The publisher must never outlive this presentation object.
+    ///
+    /// @tparam Message DSDL compiled (aka Nunavut generated) type of the message to publish.
+    /// @param subject_id The subject ID to publish the message on.
+    ///
+    template <typename Message>
+    Expected<Publisher<Message>, transport::AnyFailure> makePublisher(const transport::PortId subject_id)
```

Comment:
Yes, that is right. I should probably add such "11" example. And cover there as well similar "un-typed' Subscriber (which is coming but not in this PR yet) - that way you can run two instances of the examples and they should listen to each other.

I'll add such example (instead of currently existing commented "example_01_hello_world" one) in the next PR.